js/piwik.js

```diff
@@ -7347,6 +7346,17 @@ if (typeof window.Piwik !== 'object') {
                 });
             };
 
+            /**
+             * Sends a ping request.
+             *
+             * Ping requests do not track new actions. If they are sent within the standard visit length, they will
+             * extend the existing visit and the current last action for the visit. If after the standard visit
+             * length, ping requests will create a new visit using the last action in the last known visit.
+             */
+            this.doPing = function () {
+                trackRequest('ping=1', null, null, 'ping')
```

Comment:
This should be `this.trackRequest`

Reply:
Also I wonder if we should rename it to maybe `trackPing()` as we usually name things like this `trackGoal, trackPageview, ...` or maybe `sendPing()` as it's not really tracking anything.

Reply:
I picked `doPing` to [match the PHP API](https://developer.matomo.org/api-reference/PHP-Piwik-Tracker#doping), but could easily change it.

Reply:
Good thinking. The JS tracker and PHP tracker do have bit different naming style though. Eg PHP tracker always adds the `do` which JS tracker doesn't. Better to maybe stay consistent within JS tracker and not use `do`.

Reply:
Okay, then how about just `ping`?

Reply:
sure 👍